aten/src/ATen/native/vulkan/ops/BinaryElementwise.cpp

```diff
@@ -0,0 +1,432 @@
+#include <ATen/native/vulkan/ops/Common.h>
+#include <torch/library.h>
+
+namespace at {
+namespace native {
+namespace vulkan {
+namespace ops {
+namespace {
+
+using namespace api::utils;
+
+void check_inputs(const Tensor& input1, const Tensor& input2) {
+  TORCH_CHECK(
+      channels_size(input1) == channels_size(input2),
+      "Vulkan binary elementwise ops require channel dimension to be equal!");
+  if (batch_size(input1) != batch_size(input2)) {
+    TORCH_CHECK(
+        channels_size(input1) % 4 == 0,
+        "Vulkan binary elementwise ops require channel to be a multiple of 4 to broadcast along batch dimension!")
+  }
+  const uint32_t input1_h = height_size(input1);
+  const uint32_t input1_w = width_size(input1);
+  const uint32_t input2_h = height_size(input2);
+  const uint32_t input2_w = width_size(input2);
+  const std::string broadcast_error_msg =
+      "Incompatible input dimensions for broadcasting for Vulkan binary elementwise op!";
+  if (input1_h != input2_h) {
+    if (input1_h > input2_h) {
+      TORCH_CHECK(input2_h == 1, broadcast_error_msg);
+      TORCH_CHECK(input2_w == input1_w || input2_w == 1, broadcast_error_msg);
+    } else if (input2_h > input1_h) {
+      TORCH_CHECK(input1_h == 1, broadcast_error_msg);
+      TORCH_CHECK(input1_w == input2_w || input1_w == 1, broadcast_error_msg);
+    }
+  }
+  else if (input1_w != input2_w) {
+    if (input1_w > input2_w) {
+      TORCH_CHECK(input2_w == 1, broadcast_error_msg);
+    } else if (input2_w > input1_w) {
+      TORCH_CHECK(input1_h == 1, broadcast_error_msg);
+    }
+  }
+}
+
+bool broadcast_first_input(const vTensor& input1, const vTensor& input2) {
+  if ((input2.extents().data[1u] > 1 && input1.extents().data[1u] == 1) ||
+      (input2.extents().data[2u] > 1 && input1.extents().data[2u] == 1) ||
+      input2.extents().data[0u] > input1.extents().data[0u]) {
+    return true;
+  }
```

Comment:
You can just return the "if" condition.  if ( condition ) return true, else return false is equivalent to return condition.